packages/graphql-transformers-e2e-tests/src/__tests__/SearchableModelTransformer.e2e.test.ts

```diff
@@ -818,6 +838,87 @@ test('query for books by Agatha Christie with model using @key', async () => {
   });
 });
 
+test('test searches with datastore enabled types', async () => {
+  const createTodoResponse = await createTodo({ id: '001', name: 'get milk' });
+  expect(createTodoResponse).toBeDefined();
+  let todoName = createTodoResponse.data.createTodo.name;
+  let todoVersion = createTodoResponse.data.createTodo._version;
+
+  // Wait for the Todo to sync to Elasticsearch
+  await cf.wait(10, () => Promise.resolve());
+
+  let searchTodoResponse = await searchTodos();
+  expect(searchTodoResponse).toBeDefined();
+  expect(searchTodoResponse.data.searchTodos.items.length).toEqual(1);
+  expect(searchTodoResponse.data.searchTodos.items[0].name).toEqual(todoName);
+  expect(searchTodoResponse.data.searchTodos.items[0]._version).toEqual(todoVersion);
+  todoName = 'get soy milk';
+
+  const updateTodoResponse = await updateTodo({ id: '001', name: todoVersion, _version: todoVersion });
+  expect(updateTodoResponse).toBeDefined();
+  todoVersion += 1;
+  expect(updateTodoResponse.data.updateTodo.name).toEqual(todoName);
+  expect(updateTodoResponse.data.updateTodo._version).toEqual(todoVersion);
+
+  // Wait for the Todo to sync to Elasticsearch
+  await cf.wait(10, () => Promise.resolve());
+
+  searchTodoResponse = await searchTodos();
+  expect(searchTodoResponse.data.items.length).toEqual(1);
+  expect(searchTodoResponse.data.searchTodos.items[0].name).toEqual(todoName);
+  expect(searchTodoResponse.data.searchTodo.items[0]._version).toEqual(todoVersion);
+});
```

Comment:
would be nice to extend this test to also set _deleted to true and check that the search does not include it anymore. I think this would require updating the entry in ddb rather than through appsync though.